### PR TITLE
feat: Support `dockerPrivateKey` to specify path to SSH key

### PR DESCRIPTION
**Pull request by martinezpl**
### Problem

The SSH key that is being mounted to a dockerized pip is hardcoded as `~/.ssh/id_rsa`. This key format is considered legacy and there's no way to use a different key. Moreover, to get the user's home directory, `home` env var is being used, which does not cover 100% of cases.

### Solution

Compared to previously proposed solution of [mounting the whole .ssh directory ](https://github.com/serverless/serverless-python-requirements/pull/489), adding an option to specify the path is a lightweight approach that shouldn't be problematic. 

`process.env.HOME` is switched to `require('os').homedir()` to increase system compatibility.

The test I came up with is using a read-only deploy key to download a simple pip package from a private repo created for this purpose. Perhaps such repo should exist in /serverless profile?

Solves: https://github.com/serverless/serverless-python-requirements/issues/488, https://github.com/serverless/serverless-python-requirements/issues/617
Potentially helps: https://github.com/serverless/serverless-python-requirements/issues/560, https://github.com/serverless/serverless-python-requirements/issues/272

-edit

Some permission denied errors popped up in the second test run, not sure why?

### CODE REVIEW

**Comment by pgrzesik** [comment]
Hey @martinezpl, thanks a lot for the proposal here. I think in this particular case we could skip the integration test as it has a bit more complicated setup - otherwise we should create a repo owned by serverless to host that package to install. As for the errors - it seems like they're only happening for this PR, do you have any idea if any change introduced might be causing it?

**Comment by martinezpl** [comment]
Hi @pgrzesik! 
Regarding the errors - you can see that the first test batch has only one fail each. Two commits later (prettier + .at() -> .splice() + adding `brokenOn('win32')`)  all tests fail because of the same reason:

`EACCES: permission denied, rmdir '/home/runner/.cache/serverless-python-requirements/downloadCacheslspyc/http'`

I don't see the code-related reason here? 

As for the test, I invited you to that private repo so you can fork it to /serverless and substitute the key if you wish. This also serves as the only test for dockerSsh currently.

Otherwise, I guess it's pretty safe to assume it's gonna work if dockerizePip tests pass, as we're just appending docker commands there.

**Comment by pgrzesik** [comment]
Thanks @martinezpl - I'm not sure what is happening then - I'll dive into that probably next week as I'm pretty busy with other things, sorry about the delay

**Comment by maxpaj** [comment]
Please merge this!

**Comment by pgrzesik** [comment]
Hey @martinezpl - I believe I've fixed the CI issue, could you please rebase your branch against current `master`?

**Comment by martinezpl** [comment]
@pgrzesik permission denied error persists

**Comment by pgrzesik** [comment]
Thanks @martinezpl - I'll look into what might be causing this 😬

File: lib/pip.js [hunk 1]
```diff
@@ -275,12 +275,16 @@ async function installRequirements(targetFolder, pluginInstance) {
...
       dockerCmd.push('docker', 'run', '--rm', '-v', `${bindPath}:/var/task:z`);
       if (options.dockerSsh) {
+        const homePath = require('os').homedir();
+        const sshKeyPath =
+          options.dockerPrivateKey || `${homePath}/.ssh/id_rsa`;
+
         // Mount necessary ssh files to work with private repos
         dockerCmd.push(
           '-v',
-          `${process.env.HOME}/.ssh/id_rsa:/root/.ssh/id_rsa:z`,
```
Comment (pgrzesik) [inline lib/pip.js:281]: Could you check if that change to `os.homedir()` could be causing these issues with permissions on runner? AFAIK it still takes the result from `process.env.HOME` but nothing else in this PR looks like the cause of this problem and all other PRs are passing without problems 😕

**Comment by martinezpl** [comment]
@pgrzesik so turns out the new test is blocking all the tests. Perhaps it's some sort of a security measure related to the SSH connection to an external repo?

**Comment by pgrzesik** [comment]
Hello @martinezpl - thanks for investigating - it looks like it might be the case. In that situation, I think we should skip adding such test - could we have a simpler test that just validates proper resolution of the command when custom key is involved?

**Pull request merged**